src/reducers/trends/trendsSlice.js

```diff
@@ -43,11 +49,312 @@ export const getDefaultState = () =>
       focus: '',
       lens: 'Product',
       subLens: 'sub_product',
+      trendDepth: 5,
     },
     { ...getResetState() },
   );
 
-export const defaultTrends = getDefaultState();
+export const trendsState = getDefaultState();
+
+export const trendsSlice = createSlice({
+  name: 'trends',
+  initialState: trendsState,
+  reducers: {
+    chartTypeUpdated: {
+      reducer: (state, action) => {
+        state.chartType = state.lens === 'Overview' ? 'line' : action.payload;
+        state.tooltip = false;
+      },
+      prepare: (payload) => {
+        return {
+          payload,
+          meta: {
+            persist: PERSIST_SAVE_QUERY_STRING,
+            requery: REQUERY_NEVER,
+          },
+        };
+      },
+    },
+    dataLensChanged: {
+      reducer: (state, action) => {
+        state.subLens = '';
+        const lens = enforceValues(action.payload, 'lens');
+        switch (true) {
+          case lens === 'Company':
+            state.subLens = 'product';
+            break;
+          case lens === 'Overview':
+            state.subLens = 'product';
+            state.chartType = 'line';
+            break;
+          case lens === 'Product':
+            state.subLens = 'sub_product';
+            break;
+          default:
+            break;
+        }
+
+        state.focus = '';
+        state.lens = lens;
+        state.results = emptyResults();
+        state.tooltip = false;
+        state.trendDepth = lens === 'Company' ? 10 : 5;
+      },
+      prepare: (payload) => {
+        return {
+          payload,
+          meta: {
+            persist: PERSIST_SAVE_QUERY_STRING,
+            requery: REQUERY_HITS_ONLY,
+          },
+        };
+      },
+    },
+    dataSubLensChanged: {
+      reducer: (state, action) => {
+        return {
+          ...state,
+          subLens: action.payload.toLowerCase(),
+        };
+      },
+      prepare: (payload) => {
+        return {
+          payload,
+          meta: {
+            persist: PERSIST_SAVE_QUERY_STRING,
+            requery: REQUERY_HITS_ONLY,
+          },
+        };
+      },
+    },
+    depthChanged: {
+      reducer: (state, action) => {
+        state.trendDepth = action.payload;
+      },
+      prepare: (payload) => {
+        return {
+          payload,
+          meta: {
+            persist: PERSIST_SAVE_QUERY_STRING,
+            requery: REQUERY_HITS_ONLY,
+          },
+        };
+      },
+    },
+    depthReset: {
+      reducer: (state) => {
+        state.trendDepth = 5;
+      },
+      prepare: () => {
+        return {
+          meta: {
+            persist: PERSIST_SAVE_QUERY_STRING,
+            requery: REQUERY_HITS_ONLY,
+          },
+        };
+      },
+    },
+    focusChanged: {
+      reducer: (state, action) => {
+        const { focus, lens } = action.payload;
+        state.focus = focus;
+        state.lens = enforceValues(lens, 'lens');
+        state.tooltip = false;
```

Comment:
Since this is a boolean variable, can this be changed to `hasTooltip` `isTooltip`?

Reply:
It's actually not a bool.  It's normally an `object` for the legend on the trends tab in the external tooltip.  

We just set this to a falsy value when resetting values